core/thread.c

```diff
@@ -100,6 +101,26 @@ void thread_yield(void)
     thread_yield_higher();
 }
 
+void thread_add_to_list(list_node_t *list, thread_t *thread)
+{
+    assert (thread->status < STATUS_ON_RUNQUEUE);
+
+    uint16_t my_prio = thread->priority;
+    list_node_t *new_node = (list_node_t*)&thread->rq_entry;
+    new_node->next = NULL;
+
+    while (list->next) {
+        thread_t *list_entry = container_of((clist_node_t*)list->next, thread_t, rq_entry);
+        if (list_entry->priority > my_prio) {
```

Comment:
No, we traverse the list, which has the "highest priority" (lowest value) at the beginning. Now we skip entries until we encounter an entry which has a lower priority (higher value), and use that as our new list position.

Makes sense?